### ISSUE TITLE
I finished training, here is the evaluation result. 

### ISSUE THREAD

**Issue by nywang2019** [xavysp/DexiNed#30]
@xavysp hi, I finished training, the results are good, but the evaluation results are far from the paper.  here are the results:  Boundary
ODS: F( 0.79, 0.66 ) = 0.72   [th = 0.70]
OIS: F( 0.81, 0.68 ) = 0.74
Area_PR = 0.65
![isoF](https://user-images.githubusercontent.com/55544699/84157549-d9f7af00-aa38-11ea-9229-6831f2555ae4.png)

here are some questions: 
1. the width and height of BSDS images are 481X321, they are not divisible by 16. so I resized to 480X320 and test, and then resize the test results to 481X321 again (.jpg format).
2. I used the following matlab code for NMS:
-------------------------------------------------------
clear; clc;
path_to_pdollar_edges = 'D:\PycharmProjects2\edge-detection-DexiNed\evaluation\pdollar-edges';
path_to_pdollar_toolbox = 'D:\PycharmProjects2\edge-detection-DexiNed\evaluation\pdollar-toolbox';
path_to_input = 'D:\PycharmProjects2\edge-detection-DexiNed\evaluation\dexined-pred-f-321X481';
path_to_output = 'D:\PycharmProjects2\edge-detection-DexiNed\evaluation\nms';
addpath(genpath(path_to_pdollar_edges));
addpath(genpath(path_to_pdollar_toolbox));
%mkdir(path_to_output);
iids = dir(fullfile(path_to_input, '*.jpg'));
for i = 1:length(iids)
    edge = imread(fullfile(path_to_input, iids(i).name));
    edge=rgb2gray(edge);
    edge = 1-single(edge)/255;
    [Ox, Oy] = gradient2(convTri(edge, 4));
    [Oxx, ~] = gradient2(Ox);
    [Oxy, Oyy] = gradient2(Oy);
    O = mod(atan(Oyy .* sign(-Oxy) ./ (Oxx + 1e-5)), pi);
    % 2 for BSDS500 and Multi-cue datasets, 4 for NYUD dataset
    edge = edgesNmsMex(edge, O, 2, 5, 1.01, 8);    
    imwrite(edge, fullfile(path_to_output, [iids(i).name(1:end-4) '.png']));    
end
---------------------------------------------------
these codes are modified from RCF's edge_nms. m,  however, there are some differences.  i added `edge=rgb2gray(edge);` because the output of defined is not gray model. 
3. use bench_bsds500.m provided by BSDS to evaluate the results.

tell me if I were wrong. thanks.

**Comment by xavysp** [comment]
> @xavysp hi, I finished training, the results are good, but the evaluation results are far from the paper. here are the results: Boundary
> ODS: F( 0.79, 0.66 ) = 0.72 [th = 0.70]
> OIS: F( 0.81, 0.68 ) = 0.74
> Area_PR = 0.65
> ![isoF](https://user-images.githubusercontent.com/55544699/84157549-d9f7af00-aa38-11ea-9229-6831f2555ae4.png)
> 
> here are some questions:
> 
> 1. the width and height of BSDS images are 481X321, they are not divisible by 16. so I resized to 480X320 and test, and then resize the test results to 481X321 again (.jpg format).

Hi first of all thanks for sharing me the information. Now I am writing a paper for a journal, there I am mainly talking the disadvantage of using BSDS500 for training, because most of such image, after reprocessing to fed CNN model, the GTs are not for edge neither for boundary detection (I mean most of the GTs, so we cannot reach the state-of-the-art without fine-tuning). As you can see in the banner of DexiNed whenever we train with the right dataset  for edge detection we do not need train again in other dataset. With this what I mean is we need another dataset for boundary detection (by the way I'm working on it), or maybe improve the BSDS GT.

With respect of resizing, I did also, with 512x512. I cannot follow your results which one are yours and I suppose that are trained on BSDS I'm right?

> 2. I used the following matlab code for NMS:
> 
> ## clear; clc;
> path_to_pdollar_edges = 'D:\PycharmProjects2\edge-detection-DexiNed\evaluation\pdollar-edges';
> path_to_pdollar_toolbox = 'D:\PycharmProjects2\edge-detection-DexiNed\evaluation\pdollar-toolbox';
> path_to_input = 'D:\PycharmProjects2\edge-detection-DexiNed\evaluation\dexined-pred-f-321X481';
> path_to_output = 'D:\PycharmProjects2\edge-detection-DexiNed\evaluation\nms';
> addpath(genpath(path_to_pdollar_edges));
> addpath(genpath(path_to_pdollar_toolbox));
> %mkdir(path_to_output);
> iids = dir(fullfile(path_to_input, '_.jpg')); for i = 1:length(iids) edge = imread(fullfile(path_to_input, iids(i).name)); edge=rgb2gray(edge); edge = 1-single(edge)/255; [Ox, Oy] = gradient2(convTri(edge, 4)); [Oxx, ~] = gradient2(Ox); [Oxy, Oyy] = gradient2(Oy); O = mod(atan(Oyy ._ sign(-Oxy) ./ (Oxx + 1e-5)), pi);
> % 2 for BSDS500 and Multi-cue datasets, 4 for NYUD dataset
> edge = edgesNmsMex(edge, O, 2, 5, 1.01, 8);
> imwrite(edge, fullfile(path_to_output, [iids(i).name(1:end-4) '.png']));
> end
> these codes are modified from RCF's edge_nms. m, however, there are some differences. i added `edge=rgb2gray(edge);` because the output of defined is not gray model.
> 3. use bench_bsds500.m provided by BSDS to evaluate the results.
> 
> tell me if I were wrong. thanks.

 It's great

 Concerning to "but the evaluation results are far from the paper. here are the results:" sorry I do not follow you. I just trained DexiNed with BIPED not with BSDS, because I am just tackling edge detection.  If you need a detailed information you can write an email(you can find in my profile info). 

ahhh I set as edgesNmsMex(tmp_edge,O,1,5,1.03,8) but the difference in the results is not much.

Cheers,

**Comment by nywang2019** [comment]
sorry, it's my mistake. i trained dexined only with your dataset BIPED.  I found that the values of ODS, OIS of mine are close to your paper's.   in fact, what I am interested in is edge detection, not boundary or contour detection.  your project is interesting and useful to me.  i am trying to dig more. i will email you. thank u for your help.

**Comment by kandilidinesh** [comment]
@nywang2019 Can you help me out on calculating ODS, OIS & AP using Matlab ? I am a beginner in Matlab. I have trained the DexiNED model using my own dataset for boundary prediction in portrait images. I need to calculate the accuracy parameters.

**Comment by xavysp** [comment]
> sorry, it's my mistake. i trained dexined only with your dataset BIPED. I found that the values of ODS, OIS of mine are close to your paper's. in fact, what I am interested in is edge detection, not boundary or contour detection. your project is interesting and useful to me. i am trying to dig more. i will email you. thank u for your help.

Ok now I follow you :)  Is strange because even in the DexiNed-Pytorch the results are quite similar. In DL (if you seed the  training) you can get the same results, but I didn't so  the results will be  very similar even better, Now I am training DexiNed-TF2, the ODS and OIS are better that the results in my paper. The results in my paper are trained with the first version of BIPED so you should overcome those results. I will update the current results soon.

**Comment by nywang2019** [comment]
> @nywang2019 Can you help me out on calculating ODS, OIS & AP using Matlab ? I am a beginner in Matlab. I have trained the DexiNED model using my own dataset for boundary prediction in portrait images. I need to calculate the accuracy parameters.

i am sorry,  running evaluation code with matlab takes several hours. in my cases, nearly 8 hours. Since i have only one computer and i am very busy, i am afraid i cannot help you.
